### DISCUSSION TITLE
How to decrypt QUIC??

### DISCUSSION THREAD

**Discussion by bruno-silva-dtx** [emqx/quic#298] [Q&A] [answered]
Hello, 
My question is if there is currently any way to Quic decrypt packets, to better understand the connection, using for example wireshark, this protocol has been around for so long and I don't see how to do it?

**Comment by qzhuyan** [comment]
yes, we do.

check the `Decrypt traffic with Wireshark` part in README.md

**Reply by bruno-silva-dtx** [reply]
Yes, 

I see it now, but will this decrypt the protected payload? Or we will start to see some http3 links.
I will test and give feedback, tks

**Reply by qzhuyan** [reply]
Yes. as long as you set the SSLKEYLOGFILE file in wireshark (you could google how, it is common method for all browsers as well) and ask wireshark to 'deocde' the udps as quic, you will get decrypted payload. 

note that you must ensure you captured the handshakes at the beginning of the QUIC connection.

**Reply by bruno-silva-dtx** [reply]
Hello,
I'm sure the method works. But I'm using MQTT over QUIC, that is, I'm connecting to a bocker, what I've been trying to do is capture section keys. What I'm doing now is

export SSLKEYLOGFILE=/tmp/SSLKEYLOGFILE.log

but without results you don't capture anything.Now I'm running out of ideas, how to capture.

**Reply by qzhuyan** [reply, marked as answer]
currently we require client to set conn opt:  `{sslkeylogfile, "/tmp/SSLKEYLOGFILE"}`  (as in README) if your client use quicer.
